### PR TITLE
[SPARK-15583][SQL] Disallow altering datasource properties

### PR DESCRIPTION
**Pull request by andrewor14**
## What changes were proposed in this pull request?

Certain table properties (and SerDe properties) are in the protected namespace `spark.sql.sources.`, which we use internally for datasource tables. The user should not be allowed to

(1) Create a Hive table setting these properties
(2) Alter these properties in an existing table

Previously, we threw an exception if the user tried to alter the properties of an existing datasource table. However, this is overly restrictive for datasource tables and does not do anything for Hive tables.
## How was this patch tested?

DDLSuite

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala [hunk 1]
```diff
@@ -255,6 +255,23 @@ case class CreateDataSourceTableAsSelectCommand(
 
 
 object CreateDataSourceTableUtils extends Logging {
+
+  // TODO: Actually replace usages with these variables (SPARK-15584)
+
+  val DATASOURCE_PREFIX = "spark.sql.sources."
+  val DATASOURCE_PROVIDER = DATASOURCE_PREFIX + "provider"
+  val DATASOURCE_WRITEJOBUUID = DATASOURCE_PREFIX + "writeJobUUID"
+  val DATASOURCE_OUTPUTPATH = DATASOURCE_PREFIX + "output.path"
```
Comment (andrewor14) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala]: @dongjoon-hyun By the way I already added these here that you can use later for your patch

Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala]: Oh, I see.
Thank you!

**Comment by yhuai** [comment]
lgtm

**Comment by yhuai** [comment]
Thanks. Merging to master and branch 2.0.

**Pull request closed without merging**